File: floodFill.swift
class Solution {
    func floodFill(_ image: [[Int]], _ sr: Int, _ sc: Int, _ newColor: Int) -> [[Int]] {
     var output = image
     var stcolor = output[sr][sc]
     if stcolor == newColor {
         return output
     }   
     output[sr][sc] = newColor
     var tuples = [(1,0), (0,1), (-1,0), (0,-1)]
        
     for (x,y) in tuples {
         
         var newr = sr + x
         var newc = sc + y
         if newr >= 0 && newr < output.count && newc >= 0 && newc < output[0].count {
             if output[newr][newc] == stcolor {
                 output = floodFill(output,newr,newc,newColor)
             } 
         }
    }
    return output  
}
}

File: floodFill.py
from collections import deque
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        h, w = len(image), len(image[0])
        DIR = [(-1,0),(1,0),(0,-1),(0,1)]
        
        queue = deque()
        queue.append((sr,sc))
        
        oldColor = image[sr][sc]
        image[sr][sc] = newColor
        if oldColor != newColor:
            while queue:
                dx,dy = queue.popleft()
                for x,y in DIR:
                    r = x + dx
                    c = y + dy
                    if 0 <= r < h and 0 <= c < w and image[r][c] == oldColor:
                        queue.append((r,c))
                        image[r][c] = newColor
        return image                 
